scanpy/tools/_dendrogram.py

```diff
@@ -60,6 +63,12 @@ def dendrogram(
     ----------
     adata
         Annotated data matrix
+    groupby
+        Optional, the obs column(s) to use to group observations. Default is None.
+    dim
+        Dimension (obs or var) along which to calculate the dedrogram. Pass either `dim` or `axis` but not both.
+    axis
+        Axis (0 or 1) along which to calculate the dendrogram. Pass either `dim` or `axis` but not both.
     {n_pcs}
     {use_rep}
     var_names
```

Comment:
Is it overkill to make this also generalize i.e., you can `groupby` on either axis and you can subset (i.e., what `var_names`) on the opposite axis?

Reply:
I will say this part kind of threw me given the title of the PR.  I see what what is going on now, but would be curious your thoughts @flying-sheep.  I would expect to do identical operations on either axis but again, maybe overkill